test/ShallowWrapper-spec.js

```diff
@@ -860,6 +860,47 @@ describe('shallow', () => {
 
     });
 
+    it('should propagate events', () => {
+      const innerOnClick = sinon.spy();
+      const outerOnClick = sinon.spy();
+      class Foo extends React.Component {
+        render() {
+          return (
+            <div onClick={outerOnClick}>
+              <a onClick={innerOnClick}>foo</a>
+            </div>
+          );
+        }
+      }
+
+      const wrapper = shallow(<Foo />);
+
+      wrapper.find('a').simulate('click');
+      expect(innerOnClick.calledOnce).to.equal(true);
+      expect(outerOnClick.calledOnce).to.equal(true);
+    });
+
+    it('should respect event.stopPropagation()', () => {
+      const innerOnClick = sinon.spy(e => {
+        e.stopPropagation();
+      });
+      const outerOnClick = sinon.spy();
+      class Foo extends React.Component {
+        render() {
+          return (
+            <div onClick={outerOnClick}>
+              <a onClick={innerOnClick}>foo</a>
```

Comment:
some good additional test cases for propogation might be:

1. ensure that it can skip over parent nodes that don't have `onClick` defined, etc.
2. ensure that it propogates beyond n=2 (ie, parent + grandparent).